### DISCUSSION TITLE
Why is CQP level multiplied for AV1 NVENC encoder

### DISCUSSION THREAD

**Discussion by Bleuzen** [obsproject/obs-studio#10182] [General]
FFmpeg says the QP levels for the AV1 NVENC encoder go from `-1` to `255`, see:
`ffmpeg -hide_banner -h encoder=av1_nvenc`

However in OBS the max level in CQP mode for AV1 is `63`:
![image](https://github.com/obsproject/obs-studio/assets/12885163/24ebc47c-04e9-4a35-a4d9-26699c56d4c8)

Looking at the code, this value is later multiplied by 4:
https://github.com/obsproject/obs-studio/blob/c25f0592aeb0dce00a209e9c0906f9f70260411c/plugins/obs-ffmpeg/obs-nvenc.c#L671

Why not expose the raw value like it is done for the other codecs?

**Comment by RytoEX** [comment]
AOM and SVT-AV1 both map values 0-63 to 0-255 internally. Multiplying 0-63 by 4 gives a close enough approximation (all values are equivalent except for the mappings for 62 and 63) and, realistically, no one is or should be selecting the higher values.

https://aomedia.googlesource.com/aom/+/refs/tags/v3.8.1/av1/encoder/av1_quantize.c#895
https://gitlab.com/AOMediaCodec/SVT-AV1/-/blob/v1.8.0/Source/Lib/Encoder/Codec/EbModeDecisionProcess.h#L1278-1283

**Comment by georgedorn** [comment]
This is a massive footgun waiting to go off.

I spent many hours playing with AV1 NVENC using ffmpeg to find an ideal set of settings, and then copied those into my OBS profile and expected them to behave the same way.  Instead, my const qp setting of 125 was silently replaced with 63, which presumably was fed to NVENC as 252, which is not serviceable for much of anything.  Four hours of recording lost.

This needs UX improvement.  A warning would be nice.  Even specifying the range of -1 to 63 would have clued me into something being wrong.
